titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/KubeNotificationProcessor.java

```diff
@@ -223,15 +259,13 @@ static Task updateTaskStatus(V1Pod pod,
         } else {
             TaskStatus.Builder newStatusBuilder = TaskStatus.newBuilder().withState(TaskState.Finished);
 
-            boolean hasDeletionTimestamp = pod.getMetadata().getDeletionTimestamp() != null;
-
-            if ("failed".equalsIgnoreCase(pod.getStatus().getPhase())) {
+            if (podWrapper.getPodPhase() == PodPhase.FAILED) {
                 newStatusBuilder
-                        .withReasonCode(hasDeletionTimestamp ? REASON_TASK_KILLED : TaskStatus.REASON_FAILED)
-                        .withReasonMessage(Evaluators.getOrDefault(pod.getStatus().getMessage(), "Pod execution failed"));
+                        .withReasonCode(podWrapper.hasDeletionTimestamp() ? REASON_TASK_KILLED : TaskStatus.REASON_FAILED)
```

Comment:
same here, maybe move some of this logic into the `PodWrapper` and reuse below on line 268, e.g.: `PodWrapper#getReasonCode()`